flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SchedulerUtilsTest.java

```diff
@@ -21,17 +21,30 @@
 import org.apache.flink.api.common.JobID;
 import org.apache.flink.configuration.CheckpointingOptions;
 import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.checkpoint.CheckpointRecoveryFactory;
 import org.apache.flink.runtime.checkpoint.CompletedCheckpointStore;
-import org.apache.flink.runtime.checkpoint.StandaloneCheckpointRecoveryFactory;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.tasks.JobCheckpointingSettings;
 import org.apache.flink.util.TestLogger;
 
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
```

Comment:
We try to not use Mocking in Flink (only in very rare cases, such as external systems not providing proper interfaces). See: https://flink.apache.org/contributing/code-style-and-quality-common.html#design-for-testability

Reply:
Having read that link and the presentation I can't say I agree with said decision but you're the boss!